docs/develop/01_sdk/02_cookbook/05_messaging/01_messaging.md

```diff
@@ -0,0 +1,57 @@
+---
+id: messaging_book
+title: Generate a Message
+---
+
+import TsJsBlock from '@site/src/components/TsJsBlock';
+
+import GenerateMessage from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/messaging/01_generate_message.ts';
+import EncryptMessage from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/messaging/02_encrypt_message.ts';
+import DecryptMessage from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/messaging/03_decrypt_message.ts';
+
+KILT defines a [unicast](https://en.wikipedia.org/wiki/Unicast) [messaging protocol](../../../../concepts/06_messaging.md).
+Each of the messages sent are encrypted using the [DID key agreement key](https://www.w3.org/TR/did-core/#key-agreement).
+The key agreement key of a DID can be queried from the KILT blockchain.
+
+A message consists of the senders DID URI, the receivers DID URI, the message type and the body.
```

Comment:
```suggestion
A message consists of the sender's DID URI, the receiver's DID URI, the message type and the body.
```

Technically, a message consists of the key agreement key URIs of sender and receiver, not of their URIs.